Abschätzen der Kosten

In [1]:
import tiktoken

# Token-Kosten für das verwendete Modell (z. B. Google AI generative Modelle)
PROMPT_COST_PER_1M_TOKENS = 0.0375  # Eingabe (Prompts)
OUTPUT_COST_PER_1M_TOKENS = 0.15    # Ausgabe (Responses)

# Beispiel-Prompt
labels = [
    "Arbeitsmarktpolitik (Mindestlohn, Tarifverträge, Arbeitslosengeld)",
    "Rentenpolitik (Altersversorgung, Rentenniveau)",
    "Sozialversicherung (Kranken-, Pflegeversicherung, Unfallversicherung)",
    "Familienpolitik (Kinderbetreuung, Elternzeitregelungen, Kinderschutz, Jugendhilfe)",
    "Gleichstellung (Inklusion, Barrierefreiheit, Frauenrechte, Anti-Diskriminierung)",
    "Außenpolitik (Diplomatische, internationale Beziehungen, Friedensmissionen)",
    "Entwicklungshilfe (Internationale Armutsbekämpfung, Humanitäre Hilfe)",
    "EU-Politik (Europäische Integration, Zusammenarbeit mit EU-Institutionen)",
    "Bildungspolitik (Schulbildung, Berufsausbildung, Hochschulen)",
    "Forschung (Wissenschaftsförderung)",
    "Digitalisierung (Infrastruktur, Datenpolitik, Innovationstechnologien)",
    "Verbraucherschutz (Lebensmittelstandards, Datenschutz)",
    "Tier und Naturschutz (Biodiversität, Waldschutz, Tierrechte)",
    "Klimapolitik (Nachhaltigkeit, CO₂-Reduktion, Emissionshandel)",
    "Agrarpolitik",
    "Nukleare Sicherheit (Endlagerung von Atommüll, Sicherheit von Kernkraftwerken)",
    "Gesundheits- und Pharmapolitik (Krankenversorgung, Prävention, Arzneimittelsicherheit)",
    "Pandemiebekämpfung (Infektionsschutz, Impfstrategien)",
    "Industrie- und Handelspolitik (Freihandelsabkommen, Strukturwandel)",
    "Steuerpolitik (Einkommens- und Unternehmensbesteuerung)",
    "Haushaltspolitik (Staatsschulden und Ausgabenpolitik, Schuldenbremse)",
    "Medienpolitik (Rundfunkgesetzgebung, Pressefreiheit)",
    "Energiepolitik (Erneuerbare Energien, Effizienzsteigerungen, Versorgungssicherheit)",
    "Verkehrspolitik (Infrastruktur für Straßen, Schienen, Flugsicherheit, Schifffahrtsverkehr)",
    "Mobilitätswende (Elektroautos, öffentlicher Verkehr, öffentliche Verkehrsmittel)",
    "Kulturförderung (Denkmalschutz, Kulturelle Bildung, Sportförderung)",
    "Verteidigung (Bundeswehr, Rüstungspolitik, Cybersecurity)",
    "Migration und Integration (Asylrecht, Flüchtlingspolitik)",
    "Sicherheitsfragen (Polizei, Terrorismusbekämpfung)",
    "Gesetzgebung",
    "Verfassungsfragen (Grundrechte, Föderalismus, Menschenrechte)",
    "Stadtentwicklung (Wohnungsbau, Bauwesen, Infrastrukturprojekte)",
    "Tourismusförderung (Reisen und Gastgewerbe)"
]

rede = "diskutieren entwurf bundeshaushaltes umgesetzt erfolgen investitionen gesellschaftlichen zusammenhalt bundesfinanzminister themen benannt wohnen gleichwertige lebensverhältnisse pflege familien investitionen zukunft besonders wichtig bundeshaushalt investitionen rekordniveau erreichen fricke zahl genannt worden 40 milliarden euro glaubt fricke alleine reichen mittel einzustellen tun gemeinsam beschäftigen mittel abfließen haushalt investitionen eingestellt abgeflossen lücke fricke wenig jahren eingestellt planungsprozessen teilweise akzeptanz tun brauchen gesellschaftliche anstrengung investitionen bildet bundeshaushalt weit investitionen zukunft landes abg johannes kahrs bundeshaushalt internationalen verpflichtungen erfüllen internationale sicherheit nationale sicherheit starkes programm erfährt nato-quote entwicklungspolitik angeht leisten bekenntnis wirtschaftlichen zusammenarbeit entwicklungshaushalt erfährt oda-quote aufrechterhalten nato-quote oda-quote sicherheit entwicklung internationale verpflichtungen ganzheitlich gedacht gehört zusammen richtige pfad beschritten entwurf schulden auskommt leben zeit bewusst haushalt schulden selbstverständlichkeit schwarze null errungenschaft selbstverständlich müssen ringen ringen haushalterische nummer beiseitelegen dogma haushaltspolitik dogma kindler ausdruck generationengerechtigkeit ausdruck nachhaltigkeit fehler nachhaltigkeit ökologie definieren gehört wichtig einseitig vorgehen entsteht schieflage ökologie wirtschaft soziales gehört zusammen haushalt abbilden sieht übrigen wirtschaftliche entwicklung blick nimmt verfolgt monaten prognosen wirtschaftsentwicklung schritt schritt unten korrigiert worden korrektur prognose wirtschaftsentwicklung herausforderung bundeshaushalt planung nächsten fehlen somit milliarden euro offensichtlich unmittelbare zusammenhang guter wirtschaftlicher entwicklung mittel klimaschutz soziales familien infrastruktur wirtschaftliche entwicklung runtergeht überall problem wirtschaft mittelpunkt stellen meinung richtig olaf gesagt notwendig bestimmte schritte beispiel gefasst prozent steuerzahler wegfallen meinung schritten müssen folgen bundesfinanzminister entwurf hineingeschrieben wegfall prozent steuerzahler erster schritt meinung ersten schritt müssen folgen stillstand gerechtigkeit große aufgabe gemeinsam herangehen gemeinsam ende bringen stehen koalitionsvertrag schritt bleibt wahrung solidität haushaltsfragen maßgabe schritte unternehmensteuern sprechen wirtschaft mittelpunkt stellen personengesellschaften dürfen schlechter behandelt kapitalgesellschaften personengesellschaften entsprechen bild sozialer marktwirtschaft sehen handlungsbedarf gibt klare deutliche initiativen müssen dranbleiben wirtschaft wichtig wirtschaft hängt bereichen finanzieren umsetzen reden besonderer weise fragen klimaschutzes angesprochen worden globale herausforderung pro-kopf-ausstoß doppelt hoch internationale schnitt national besondere verantwortung lücke schließen zielen erreicht besondere aufgabe geiste angehen ausdruck kommt gesagt land ingenieuren land technologie setzt land innovativen mittelstand industrie vorne denkt technologische lösungen entwickelt innovation setzen ende erfolgreich ganze globale weltweit gekuckt deutschen konsequenten klimaschutz individuelle mobilität ausdruck freiheit erhalten arbeitsplätze wohlstand erhalten gelingt folgen erfüllen globale aufgabe sinnvoll innovationen müssen mittelpunkt komme haushalt zusammenzubringen klimapolitik große priorität verbinden haushalt sagt generalüberprüfung brauchen check kopf fuß stellen dürfen on top kommen stellen müssen aufgaben vordringlich generelle überprüfung prioritätensetzung bringt zeiten schwieriger kassenlage enger bedeutet automatisch schulden heißt prioritätensetzung heißt effizienz gilt energie gilt bereichen müssen fragen maßnahme kostet bringt erreichen größtmöglichen erfolg eingesetzten mittel aufgabe stehen selbstverständlich klimaschutz herausragendes thema co2-ausstoß maßstab lenken senken gelingen selbstverständlich bedeutet verursacht egal bereich bezahlen co2 verursacht einspart entlastet linie umsetzen soziale balance gewahrt wirtschaftlich erfolgreich bleiben besondere verantwortung akzeptieren bundesfinanzminister unterschiedliche ansätze vorschläge instrumente gibt stehen gemeinsam herausforderung klimapolitik aufzuforsten mag unterschiedliche herangehensweisen sicher ende guten mischwald zusammenführen mischwälder nachhaltigsten aufgabe steht stehen insgesamt möchte ausdrücklich gedanken nationalen konsenses klimapakt aufnehmen passieren darf maßnahmen beschließen umgesetzt nehme steuerliche förderung gebäudesanierung maßnahme effizient klima gut handwerk gut konjunktur natur zusammenkommen beschlossen föderalen mühlen bundesrat zerrieben worden reden schlafenden riesen wärmemarkts schlummert darf passieren bund beschließen kommt hängepartien gibt wäre wichtig denken hinaus notwendig denken generation beschlüsse dürfen halbwertzeiten legislaturperioden müssen ambitioniert vorangehen weiterdenken sinne breiten konsenses richtig arbeiten gilt bundeshaushalt besonderer weise nächster"

prompt_template = """
Klassifiziere die folgende Bundestagsrede entsprechend der folgenden Themen: {labels}.
Deine Antwort darf NUR aus einem Thema bestehen und soll GENAU GLEICH geschrieben werden, wie eines der gelisteten T.
Rede: "{rede}"
"""


prompt = prompt_template.format(labels=", ".join(labels), rede=rede)

# Token-Zählung mit tiktoken
def count_tokens(prompt, model="gpt-3.5-turbo"):  # Token-Modell ähnlich wie bei Google AI
    encoding = tiktoken.encoding_for_model(model)
    tokens = encoding.encode(prompt)
    return len(tokens)

# Tokens berechnen
num_prompt_tokens = count_tokens(prompt)
num_output_tokens = 50  # Annahme: durchschnittlich 50 Tokens in der Antwort

# Kostenberechnung
prompt_cost = (num_prompt_tokens / 1_000_000) * PROMPT_COST_PER_1M_TOKENS
output_cost = (num_output_tokens / 1_000_000) * OUTPUT_COST_PER_1M_TOKENS
total_cost = prompt_cost + output_cost 
total_prompts = 25000

# Ergebnisse ausgeben
print(f"Prompt-Tokens: {num_prompt_tokens}")
print(f"Output-Tokens: {num_output_tokens}")
print(f"Prompt-Kosten: ${prompt_cost * total_prompts}")
print(f"Output-Kosten: ${output_cost * total_prompts}")
print(f"Gesamtkosten pro Labeling Vorgang: ${total_cost * total_prompts}")


Prompt-Tokens: 2025
Output-Tokens: 50
Prompt-Kosten: $1.8984375
Output-Kosten: $0.1875
Gesamtkosten pro Labeling Vorgang: $2.0859375


Ergebnis: Realisierbar

Prompt-Tokens: 2025
Output-Tokens: 50
Prompt-Kosten: $1.8984375
Output-Kosten: $0.1875
Gesamtkosten pro Labeling Vorgang: $2.0859375

Labeling Vorgang: 

In [ ]:
import os
import pandas as pd
from tqdm import tqdm
from dotenv import load_dotenv
import google.generativeai as genai

# .env-Datei laden
load_dotenv()

# Google API-Schlüssel laden
api_key = os.getenv("GOOGLE_API_KEY")
if not api_key:
    raise ValueError("API-Schlüssel nicht gefunden. Bitte in der .env-Datei angeben.")
genai.configure(api_key=api_key)

# Initialisiere das Modell
model = genai.GenerativeModel("gemini-1.5-flash-8b")

# Pfade zu den CSV-Dateien
input_csv_path = "v2_2_4_final_data_cleaning.csv"
output_csv_path = "v4_1_labeled.csv"

# Liste der Labels
labels = [
    "Arbeitsmarktpolitik (Mindestlohn, Tarifverträge, Arbeitslosengeld)",
    "Rentenpolitik (Altersversorgung, Rentenniveau)",
    "Sozialversicherung (Kranken-, Pflegeversicherung, Unfallversicherung)",
    "Familienpolitik (Kinderbetreuung, Elternzeitregelungen, Kinderschutz, Jugendhilfe)",
    "Gleichstellung (Inklusion, Barrierefreiheit, Frauenrechte, Anti-Diskriminierung)",
    "Außenpolitik (Diplomatische, internationale Beziehungen, Friedensmissionen)",
    "Entwicklungshilfe (Internationale Armutsbekämpfung, Humanitäre Hilfe)",
    "EU-Politik (Europäische Integration, Zusammenarbeit mit EU-Institutionen)",
    "Bildungspolitik (Schulbildung, Berufsausbildung, Hochschulen)",
    "Forschung (Wissenschaftsförderung)",
    "Digitalisierung (Infrastruktur, Datenpolitik, Innovationstechnologien)",
    "Verbraucherschutz (Lebensmittelstandards, Datenschutz)",
    "Tier und Naturschutz (Biodiversität, Waldschutz, Tierrechte)",
    "Klimapolitik (Nachhaltigkeit, CO₂-Reduktion, Emissionshandel)",
    "Agrarpolitik",
    "Nukleare Sicherheit (Endlagerung von Atommüll, Sicherheit von Kernkraftwerken)",
    "Gesundheits- und Pharmapolitik (Krankenversorgung, Prävention, Arzneimittelsicherheit)",
    "Pandemiebekämpfung (Infektionsschutz, Impfstrategien)",
    "Industrie- und Handelspolitik (Freihandelsabkommen, Strukturwandel)",
    "Steuerpolitik (Einkommens- und Unternehmensbesteuerung)",
    "Haushaltspolitik (Staatsschulden und Ausgabenpolitik, Schuldenbremse)",
    "Medienpolitik (Rundfunkgesetzgebung, Pressefreiheit)",
    "Energiepolitik (Erneuerbare Energien, Effizienzsteigerungen, Versorgungssicherheit)",
    "Verkehrspolitik (Infrastruktur für Straßen, Schienen, Flugsicherheit, Schifffahrtsverkehr)",
    "Mobilitätswende (Elektroautos, öffentlicher Verkehr, öffentliche Verkehrsmittel)",
    "Kulturförderung (Denkmalschutz, Kulturelle Bildung, Sportförderung)",
    "Verteidigung (Bundeswehr, Rüstungspolitik, Cybersecurity)",
    "Migration und Integration (Asylrecht, Flüchtlingspolitik)",
    "Sicherheitsfragen (Polizei, Terrorismusbekämpfung)",
    "Gesetzgebung",
    "Verfassungsfragen (Grundrechte, Föderalismus, Menschenrechte)",
    "Stadtentwicklung (Wohnungsbau, Bauwesen, Infrastrukturprojekte)",
    "Tourismusförderung (Reisen und Gastgewerbe)"
]

# Prompt-Vorlage für das Modell
prompt_template = """
Klassifiziere die folgende Bundestagsrede entsprechend der folgenden Themen: {labels}.
Deine Antwort darf NUR aus einem Thema bestehen und soll GENAU GLEICH geschrieben werden, wie eines der gelisteten T.
Rede: "{rede}"
"""

# Daten laden
df = pd.read_csv(input_csv_path)

# Sicherstellen, dass die Spalte 'label' existiert
if "label" not in df.columns:
    df["label"] = None

# Startindex für die Verarbeitung ermitteln
start_index = df[df["label"].isna()].index.min()

if start_index is None:
    print("Alle Zeilen sind bereits gelabelt. Verarbeitung beendet.")
else:
    print(f"Beginne mit der Verarbeitung ab Zeile {start_index}.")

# Verarbeitung der Reden mit Fortschrittsanzeige
for index, row in tqdm(df.iloc[start_index:].iterrows(), total=df.shape[0] - start_index, desc="Verarbeitung der Reden"):
    rede = row["text"]

    # Aktuelle Statusanzeige
    print(f"\n=== Verarbeite Zeile {index} ===")
    print(f"Text der Rede: {rede[:100]}...")  # Nur die ersten 100 Zeichen anzeigen

    # Prompt für die aktuelle Rede erstellen
    prompt = prompt_template.format(labels=", ".join(labels), rede=rede)

    # Anfrage an Google Generative AI
    try:
        print("Sende Anfrage an Google Generative AI...")
        response = model.generate_content(prompt)
        label = response.text.strip()
        print(f"Antwort von Google Generative AI: {label}")
        df.at[index, "label"] = label

    except Exception as e:
        print(f"Fehler bei der Verarbeitung der Rede in Zeile {index}: {e}")
        df.at[index, "label"] = "Fehler"

    # Zwischenspeicherung nach 50 Iterationen
    if (index - start_index + 1) % 50 == 0:
        df.to_csv(output_csv_path, index=False)
        print(f"Zwischenspeicherung nach {index - start_index + 1} Zeilen.")

# Endgültige Speicherung
df.to_csv(output_csv_path, index=False)
print(f"Die Datei mit den Labels wurde gespeichert: {output_csv_path}")


In [ ]:
import os
import pandas as pd
from tqdm import tqdm
from dotenv import load_dotenv
import google.generativeai as genai

# .env-Datei laden
load_dotenv()

# Google API-Schlüssel laden
api_key = os.getenv("GOOGLE_API_KEY")
if not api_key:
    raise ValueError("API-Schlüssel nicht gefunden. Bitte in der .env-Datei angeben.")
genai.configure(api_key=api_key)

# Initialisiere das Modell
model = genai.GenerativeModel("gemini-1.5-flash-8b")

# Pfade zu den CSV-Dateien
input_csv_path = "v4_1_additionaly_labeled.csv"
output_csv_path = "v4_1_additionaly_labeled_final.csv"

# Liste der Labels
labels = [
    "Arbeitsmarktpolitik (Mindestlohn, Tarifverträge, Arbeitslosengeld)",
    "Rentenpolitik (Altersversorgung, Rentenniveau)",
    "Sozialversicherung (Kranken-, Pflegeversicherung, Unfallversicherung)",
    "Familienpolitik (Kinderbetreuung, Elternzeitregelungen, Kinderschutz, Jugendhilfe)",
    "Gleichstellung (Inklusion, Barrierefreiheit, Frauenrechte, Anti-Diskriminierung)",
    "Außenpolitik (Diplomatische, internationale Beziehungen, Friedensmissionen)",
    "Entwicklungshilfe (Internationale Armutsbekämpfung, Humanitäre Hilfe)",
    "EU-Politik (Europäische Integration, Zusammenarbeit mit EU-Institutionen)",
    "Bildungspolitik (Schulbildung, Berufsausbildung, Hochschulen)",
    "Forschung (Wissenschaftsförderung)",
    "Digitalisierung (Infrastruktur, Datenpolitik, Innovationstechnologien)",
    "Verbraucherschutz (Lebensmittelstandards, Datenschutz)",
    "Tier und Naturschutz (Biodiversität, Waldschutz, Tierrechte)",
    "Klimapolitik (Nachhaltigkeit, CO₂-Reduktion, Emissionshandel)",
    "Agrarpolitik",
    "Nukleare Sicherheit (Endlagerung von Atommüll, Sicherheit von Kernkraftwerken)",
    "Gesundheits- und Pharmapolitik (Krankenversorgung, Prävention, Arzneimittelsicherheit)",
    "Pandemiebekämpfung (Infektionsschutz, Impfstrategien)",
    "Industrie- und Handelspolitik (Freihandelsabkommen, Strukturwandel)",
    "Steuerpolitik (Einkommens- und Unternehmensbesteuerung)",
    "Haushaltspolitik (Staatsschulden und Ausgabenpolitik, Schuldenbremse)",
    "Medienpolitik (Rundfunkgesetzgebung, Pressefreiheit)",
    "Energiepolitik (Erneuerbare Energien, Effizienzsteigerungen, Versorgungssicherheit)",
    "Verkehrspolitik (Infrastruktur für Straßen, Schienen, Flugsicherheit, Schifffahrtsverkehr)",
    "Mobilitätswende (Elektroautos, öffentlicher Verkehr, öffentliche Verkehrsmittel)",
    "Kulturförderung (Denkmalschutz, Kulturelle Bildung, Sportförderung)",
    "Verteidigung (Bundeswehr, Rüstungspolitik, Cybersecurity)",
    "Migration und Integration (Asylrecht, Flüchtlingspolitik)",
    "Sicherheitsfragen (Polizei, Terrorismusbekämpfung)",
    "Gesetzgebung",
    "Verfassungsfragen (Grundrechte, Föderalismus, Menschenrechte)",
    "Stadtentwicklung (Wohnungsbau, Bauwesen, Infrastrukturprojekte)",
    "Tourismusförderung (Reisen und Gastgewerbe)"
]

# Prompt-Vorlage für die Label-Überprüfung und Stichwortgenerierung
review_prompt_template = """
Die folgende Bundestagsrede wurde als {label} klassifiziert. Generiere nochmal alternative Klassifizierungen. Hier siehst du noch einmal mögliche andere Labels:
{labels}
Generiere zusätzlich 5 Schlüsselwörter, die den Inhalt der Rede näher beschreiben.

Rede: "{rede}"

Das Ergebniss muss zwingend wie folgt formatiert sein:
[Label1, alternatives Label2, alternatives Label3, Schlüsselwört1, Schlüsselwört2, Schlüsselwört3, Schlüsselwört4, Schlüsselwört5]
"""

# Daten laden
df = pd.read_csv(input_csv_path)

# Sicherstellen, dass die Spalten 'label' und 'output' existieren
if "label" not in df.columns:
    raise ValueError("Die Spalte 'label' fehlt. Bitte die erste Iteration durchführen.")


# Startindex für die Verarbeitung ermitteln
start_index = df[df["alt_labels"].isna()].index.min()

if start_index is None:
    print("Alle Zeilen wurden bereits überprüft. Verarbeitung beendet.")
else:
    print(f"Beginne mit der Überprüfung ab Zeile {start_index}.")

# Verarbeitung der Reden mit Fortschrittsanzeige
for index, row in tqdm(df.iloc[start_index:].iterrows(), total=df.shape[0] - start_index, desc="Überprüfung der Labels"):
    if pd.isna(row["alt_labels"]):
        rede = row["rede"]
        label = row["label"]

        # Prompt für die aktuelle Rede erstellen
        review_prompt = review_prompt_template.format(label=label, rede=rede, labels=labels)

        # Anfrage an Google Generative AI
        try:
            response = model.generate_content(review_prompt)
            output = response.text.strip()
            df.at[index, "alt_labels"] = output
            print(f"Antwort von Google Generative AI (Zeile {index}): {output}")

        except Exception as e:
            print(f"Fehler bei der Überprüfung der Rede in Zeile {index}: {e}")
            df.at[index, "alt_labels"] = "[Fehler]"

        # Zwischenspeicherung nach 50 Iterationen
        if (index - start_index + 1) % 50 == 0:
            df.to_csv(output_csv_path, index=False)
            print(f"Zwischenspeicherung nach {index - start_index + 1} Zeilen.")

# Endgültige Speicherung
df.to_csv(output_csv_path, index=False)
print(f"Die Datei mit den überprüften Labels wurde gespeichert: {output_csv_path}")



Final Touches: Dataset

In [1]:
import pandas as pd

# Pfad zur CSV-Datei
csv_path = "v4_1_additionaly_labeled.csv"

# Daten laden
df = pd.read_csv(csv_path)


# Spalten umbenennen
df.rename(columns={
    "unique_id": "id",
    "id": "sitzungs_id",
    "dokumentnummer": "dokumenten_id",
    "datum": "datum",
    "Dr.": "titel",
    "Sprecher": "redner",
    "Partei": "partei",
    "Amt": "amt",
    "label": "label",
    "output": "alt_labels",
    "text": "rede"
}, inplace=True)

# Spalten entfernen
df.drop(columns=["amt", "titel"], inplace=True)

# Spaltenreihenfolge anpassen
df = df[["id", "sitzungs_id", "dokumenten_id", "datum", "redner", "partei", "label", "alt_labels", "rede"]]

# Daten speichern
df.to_csv(csv_path, index=False)
print(f"Das angepasste Dataset wurde gespeichert: {csv_path}")

Das angepasste Dataset wurde gespeichert: v4_1_additionaly_labeled.csv


Die Labels auf den Volltext übertragen: 

In [2]:
import pandas as pd
# Pfad zur CSV-Datei mit den separaten Reden
csv_path = "v4_1_S_additionaly_labeled_final_filtered.csv"
unique_speeches_csv_path = "v2_2_1_unique_separat_speeches.csv"

df1 = pd.read_csv(csv_path)
df2 = pd.read_csv(unique_speeches_csv_path)

# Merge der DataFrames auf Basis der Spalte 'id'
merged_df = pd.merge(df1.drop(columns=['rede']), df2[['id', 'text']], on='id', how='left')

# Ergebnis speichern
output_csv_path = "v0_unique_separat_speeches_labeled.csv"
merged_df.to_csv(output_csv_path, index=False)
print(f"Das zusammengeführte Dataset wurde gespeichert: {output_csv_path}")

Das zusammengeführte Dataset wurde gespeichert: v0_unique_separat_speeches_labeled.csv


LLM Halluzinations: hier und da folgt das LLM nicht dem gegeben Antwortschema. Die Zeilen erkennt man weil verhältnissmäßig sehr viel text vorkommt

In [3]:
import pandas as pd
# Pfad zur CSV-Datei mit den separaten Reden
csv_path = "v4_1_additionaly_labeled_final.csv"
df = pd.read_csv(csv_path)


# Zeilen löschen, die mehr als 600 Zeichen in der Spalte 'alt_labels' haben
#df = df[df['alt_labels'].str.len() <= 600]
df = df[df['alt_labels'].str.count('\[') <= 1]
df = df[df['alt_labels'].str.count('\]') <= 1]

def check_columns(row):
    return len(row) > 9

df = df[~df.apply(check_columns, axis=1)]

# Ergebnis speichern
df.to_csv('v4_1_additionaly_labeled_final_filtered.csv', index=False)
print(f"Das gefilterte Dataset wurde gespeichert: v4_1_additionaly_labeled_final_filtered.csv")


Das gefilterte Dataset wurde gespeichert: v4_1_additionaly_labeled_final_filtered.csv
